components/higher-order/with-dnd/index.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { DragDropContext, DragSource, DropTarget } from 'react-dnd';
+import HTML5Backend from 'react-dnd-html5-backend';
+
+/**
+ * WordPress dependencies
+ */
+import { Component } from '@wordpress/element';
+
+/* Testing... */
+const ModifiedBackend = ( ...args ) => {
+    const instance = new HTML5Backend( ...args );
+
+    const listeners = [
+        'handleTopDragStart',
+        'handleTopDragStartCapture',
+        'handleTopDragEndCapture',
+        'handleTopDragEnter',
+        'handleTopDragEnterCapture',
+        'handleTopDragLeaveCapture',
+        'handleTopDragOver',
+        'handleTopDragOverCapture',
+        'handleTopDrop',
+        'handleTopDropCapture',
+        'handleSelectStart',
+    ];
+
+    const shouldIgnoreTarget = ( target ) => {
+        return target.className.split(' ').indexOf( 'is-reordering-in-progress' );
```

Comment:
This logic doesn't do as you might expect. I assume you're checking whether the target has the specified class assigned, but `Array#indexOf` returns `-1` if the item is not in the array. This is a [truthy](https://developer.mozilla.org/en-US/docs/Glossary/Truthy) value; conversely, `0` is a valid return value but is [falsy](https://developer.mozilla.org/en-US/docs/Glossary/Falsy).

Instead, you might consider:

- Comparing against `-1`
   - `-1 !== target.className.split( ' ' ).indexOf( 'is-reordering-in-progress' );`
- Using [Lodash's `_.includes`](https://lodash.com/docs/4.17.4#includes) which is a bit more semantic to what you're trying to test
   - `includes( target.className.split( ' ' ).indexOf( 'is-reordering-in-progress' );`
- Using `Element#classList.contains` which is designed for what you're trying to do here:
   - `target.classList.contains( 'is-reordering-in-progress' );`

Reply:
There's a styling issue here: there needs to be a space between the parentheses of `split`, i.e. `split( ' ' )`

You might consider installing an ESLint plugin for your editor to help surface these styling issues more apparently:

https://eslint.org/docs/user-guide/integrations

Reply:
Thanks @aduth for your comments and pointers. This part of the code is actually not used anymore, as I have instead found a patch that fixes the issues I was trying to get to here. I will send an update soon, so apologies for any inconvenience. 

I may actually refrain from using react-dnd altogether. I am not too happy with all the clutter that is created. I am running a separate experiment right now with the existing drop-zones. I will share soon.